### ISSUE TITLE
Database files are incompatible with server after pulling new version of bacularis-standalone image

### ISSUE THREAD

**Issue by kibeb** [bacularis/bacularis-docker#8]
How am I supposed to upgrade Postgres database files inside my docker container after pulling new version of bacularis-standalone image?
Shouldn't this be handled automatically?
TIA.

```
waiting for server to start....2024-10-14 04:28:12.291 UTC [9] FATAL:  database files are incompatible with server
2024-10-14 04:28:12.291 UTC [9] DETAIL:  The data directory was initialized by PostgreSQL version 14, which is not compatible with this version 16.3.
 stopped waiting
pg_ctl: could not start server
Examine the log output.
```

**Comment by ganiuszka** [comment]
Hello @kibeb,

Thanks for your report and ideas.

It is true that something has changed in the database format. We informed in the release notes ``4.0.0`` that we did major upgrade of the PostgreSQL server and that it requires manual running ``pg_upgrade``.

Upgrading the database data after the database format change is not a trivial task that requires additional attention, specially for big databases with big data. I think it can be the reason why the PostgreSQL and Bacula system package maintainers didn't take it on theirs shoulders but left it to manual running. In the Bacularis Project we do the same and we ask about maintaining it manually.

Best regards,
Marcin Haba (gani)
